### PR TITLE
Fine-tuning and running QA with bert for our problem 

### PR DESCRIPTION
**Pull request by SzymonKozl**

### CODE REVIEW

**Comment by SzymonKozl** [comment]
At this moment this code is WIP

File: research/question_answering_approach/ds/18929485529.csv
Comment (GustawB) [inline research/question_answering_approach/ds/18929485529.csv:1]: didn't we discuss with Murmuras at some point that this data should not be shared (this repo is public)?

**Review by GustawB — CHANGES_REQUESTED** [review]
Looks alright, but I think it would be nice if you:
1. Added a short intro describing why are you researching this approach
2. There isn't much description of what's going on throught the file; while in most places it's non-problematic, at some point in notebook you get some kind of result and say that "it's bad". After you explanation on weekly today I know why, but without it, all I see is a low score and some  start and end indexes. I would elaborate on what does this result represent and then why this particular one is bad.

You can also elaborate somewhere (ig at the beginning) what QA does (this connects with describing why you research it, but it's important because it will also make it more clear why e.g. results you obtain are good/bad).

File: research/question_answering_approach/ds/18929485529.csv
Comment (SzymonKozl) [reply research/question_answering_approach/ds/18929485529.csv:1]: removed

File: research/question_answering_approach/qa_finetuning.ipynb [hunk 1]
```diff
@@ -0,0 +1,953 @@
...
+    "QA is machine learning task focused on retrieving answers for questions from given text context. In this notebook I will experiment with extractive QA, architecture in which model predicts fragment of input context in which response is. It is worth noticing that there are different approaches to QA task, like generative QA which allows model to generate response that might not be explicit part of context.\n",
+    "### Why we would want to use this?\n",
+    "Our task seems similar to extractive QA. There are some limitations to be addressed, for product name being split with some xml tag like\n",
+    "```xml\n",
+    "śmiej<bold>żelki</bold>\n",
+    "```\n",
+    "that can make it impossible to select single fragment containing answer to question \"What is name of product?\" but in general this approach seems suitable.\n",
+    "### Is it possible?\n",
+    "tldr: idk </br>\n",
+    "I did not find and fine-tunes of BERT on xml or even html. Although there are llama models finetuned for RAG, task similar to QA working with HTML data (like zstanjj/HTML-Pruner-Llama-1B).\n"
```
Comment (GustawB) [inline research/question_answering_approach/qa_finetuning.ipynb:27]:
```suggestion
    "I did not find or fine-tuned any BERT model on xml or even html. Although there are llama models finetuned for RAG, task similar to QA working with HTML data (like zstanjj/HTML-Pruner-Llama-1B).\n"
```

File: research/question_answering_approach/qa_finetuning.ipynb [hunk 2]
```diff
@@ -0,0 +1,953 @@
...
+    "### What is QA task\n",
+    "QA is machine learning task focused on retrieving answers for questions from given text context. In this notebook I will experiment with extractive QA, architecture in which model predicts fragment of input context in which response is. It is worth noticing that there are different approaches to QA task, like generative QA which allows model to generate response that might not be explicit part of context.\n",
+    "### Why we would want to use this?\n",
+    "Our task seems similar to extractive QA. There are some limitations to be addressed, for product name being split with some xml tag like\n",
+    "```xml\n",
+    "śmiej<bold>żelki</bold>\n",
+    "```\n",
+    "that can make it impossible to select single fragment containing answer to question \"What is name of product?\" but in general this approach seems suitable.\n",
+    "### Is it possible?\n",
+    "tldr: idk </br>\n",
```
Comment (GustawB) [inline research/question_answering_approach/qa_finetuning.ipynb:26]: Hmm, not sure if I understand; can we talk irl about this?

**Review by GustawB — CHANGES_REQUESTED** [review]

File: research/question_answering_approach/qa_finetuning.ipynb (hunk 1 above)
Comment (SzymonKozl) [reply research/question_answering_approach/qa_finetuning.ipynb:27]: This correction changes meaning of this sentence. I propose rather:
```suggestion
    "I did not find any fine-tuned versions of BERT on xml or even html. Although there are llama models finetuned for RAG, task similar to QA working with HTML data (like zstanjj/HTML-Pruner-Llama-1B).\n"
```

**Review by GustawB — APPROVED** [review]

**Pull request merged**